psi4/src/psi4/libfunctional/superfunctional.cc

```diff
@@ -78,7 +80,10 @@ void SuperFunctional::common_init() {
 }
 std::shared_ptr<SuperFunctional> SuperFunctional::blank() { return std::make_shared<SuperFunctional>(); }
 std::shared_ptr<SuperFunctional> SuperFunctional::XC_build(std::string name, bool unpolarized) {
-    // Only allow build from full XC kernals
+    // Capitalize name for consistency
+    std::transform(name.begin(), name.end(), name.begin(), ::toupper);
+
```

Comment:
We have various string functions in `psi4/libpsi4util/libpsi4util.h`
here `void to_upper(std::string &str) { std::transform(str.begin(), str.end(), str.begin(), ::toupper); }`  should do the trick.

Reply:
I was merely following Psi4 standards
```
$ grep toupper -R src/|grep std::transform
src/export_mints.cc:            std::transform(symbol.begin(), symbol.end(), symbol.begin(), ::toupper);
src/export_mints.cc:            std::transform(label.begin(), label.end(), label.begin(), ::toupper);
src/export_mints.cc:            std::transform(symbol.begin(), symbol.end(), symbol.begin(), ::toupper);
src/export_mints.cc:            std::transform(label.begin(), label.end(), label.begin(), ::toupper);
src/psi4/libfunctional/superfunctional.cc:    std::transform(name.begin(), name.end(), name.begin(), ::toupper);
src/psi4/libmints/basisset.cc:    std::transform(upper.begin(), upper.end(), upper.begin(), ::toupper);
src/psi4/libmints/basisset.cc:    std::transform(nameUpperCase.begin(), nameUpperCase.end(), nameUpperCase.begin(), ::toupper);
src/psi4/libpsio/filescfg.cc:    std::transform(fkwd.begin(), fkwd.end(), fkwd.begin(), static_cast<int (*)(int)>(toupper));
src/psi4/libplugin/load_plugin.cc:    std::transform(info.name.begin(), info.name.end(), info.name.begin(), [](unsigned char c) { return std::toupper(c); });
src/psi4/libpsi4util/stl_string.cc:void to_upper(std::string &str) { std::transform(str.begin(), str.end(), str.begin(), ::toupper); }
src/psi4/liboptions/liboptions.cc:void DataType::to_upper(std::string& str) const { std::transform(str.begin(), str.end(), str.begin(), ::toupper); }
src/psi4/liboptions/liboptions.cc:void Options::to_upper(std::string& str) const { std::transform(str.begin(), str.end(), str.begin(), ::toupper); }
src/create_new_plugin.cc:        std::transform(format_PLUGIN.begin(), format_PLUGIN.end(), format_PLUGIN.begin(), ::toupper);
src/core.cc:    std::transform(nonconst_key.begin(), nonconst_key.end(), nonconst_key.begin(), ::toupper);
```

Reply:
I can change all of these to `to_upper`